Framework/API/test/ExperimentInfoTest.h

```diff
@@ -901,14 +907,9 @@ class ExperimentInfoTest : public CxxTest::TestSuite {
     instrument->markAsSamplePos(sample);
 
     ExperimentInfo expInfo;
-    expInfo.setInstrument(instrument);
-    const Mantid::API::ComponentInfo &compInfo = expInfo.componentInfo();
-
-    TSM_ASSERT_EQUALS("Should be a valid component index", 0,
-                      compInfo.indexOf(det1->getComponentID()));
-    TSM_ASSERT_THROWS("Should throw. Duplicate should have been rejected",
-                      compInfo.indexOf(det2->getComponentID()),
-                      std::out_of_range &);
+    TSM_ASSERT_THROWS(
+        "Component count and detector count will be out of sync. Should throw.",
+        expInfo.setInstrument(instrument), std::invalid_argument &);
```

Comment:
Good point. But I'm going to leave this for now. I need to explore how strict we can be about dealing with this kind of error condition. I'm going to lean on the system tests as part of this PR to get some kind of indication.